quesma/v2/core/diag/diagnostic.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package diag
+
+// Diagnostic is an interface that provides access to diagnostic tools.
+// Content of this interface may change in the future.
+type Diagnostic interface {
+	PhoneHomeAgent() PhoneHomeClient
```

Comment:
I think that at some point in time `PhoneHomeAgent` and `DebugInfoCollector` might be combined together

Reply:
I will wait for the rest of the components. The scope of utility components will be discovered in the future ;)